src/spikeinterface/core/tests/test_waveform_tools.py

```diff
@@ -52,96 +54,88 @@ def test_waveform_tools():
     unit_ids = sorting.unit_ids
 
     some_job_kwargs = [
-        {},
         {"n_jobs": 1, "chunk_size": 3000, "progress_bar": True},
         {"n_jobs": 2, "chunk_size": 3000, "progress_bar": True},
     ]
```

Comment:
Could you use parameterize for this tests instead of the triple loop for the kwargs inside the test? I will make the code simpler to read (not as nested) and it will also make the tests independent and it will be simpler to spot which one is failing?

Reply:
I want to debug and stop in a spcific part of the code for a specific example and I am not able to with the current configuration.

Reply:
I think the test was already like this.
Sometimes it is cools to also have loops when you share some variable this avoid long pytest.fixture chain + pytest.parametrize storie.
I think this is a matter of taste no ?

Reply:
To some deegree it is, @samuelgarcia. Specially for using many fixtures I find myself agreeing with some of your taste arguments (it looks like a lot of clutter) and I have a lot of sympathy for your own usability concerns (adds more boilerplate to your way of debugging by using a main call at the end of the testing file, what a pain). For the latter I tried to help you writing that boilerplate myself for some of the functionst that I have modified.

Maybe we can reach a compromise if you can suggest some workarounds for my debuggin workflow? 
I really want to stop and debugg a specific line with some set of parameters. I don't know of a simple way of doing it this way. I would need to add an if block to check that the parameters are the ones that I want to look at and then set the traces of the debugger to that new if line which is a cumbersome, error prone and labrious task.

With parameterization I can just click a line (even from my IDE) and run the specific mode that I want to test or debugg:
![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/14fbae8f-77d1-42c5-982a-51e9878005db)
(see on the left that I can just run the specific test that I want)

If you really dislike parameterize maybe we could have one loop for test or break them appart in some other way? That will make my life easier and it will cost you only some duplication of code (in practice copy paste).  Plus, it will make the tests closer in spirit to the ideal of unit testing. I am open to other suggestions but I would like you to recognize my troubles the way that I recognize yours.

I would also like to point out that my debugging workflow by setting traces is more standard than yours. It is supported in the [standard library](https://docs.python.org/3/library/pdb.html) and, as you can see above, it is already well integrated in IDEs like VScode. Future developers are more likely to be using that type of workflow. 

Plus, using standard tooling of pytest like parameterize opens the doors to certain futures that we don't have available right now. Two examples: 1) The reporting tool that I built for the CI to see which tests take more time is more useful when the tests are clearly separated. Right now, all the time taken by the tests is lumped into one thing and we can't compare paramters choices to get a quick sense if something runs faster and slower in other architectures for example 2) It does not apply here but there are packages like [xdist](https://pypi.org/project/pytest-xdist/) that we could use to improve the running times of the CI and they work better when the tests are more atomic.

Reply:
OK: one point for you.
I will do this next time I promise.
But I need this in the main quite soon and I am quick busy....

Reply:
The `_check_all_wf_equal()` at the end the tripple loop prevent to refactor this using pytest_parametrize